### DISCUSSION TITLE
Ability to save emails without attachments

### DISCUSSION THREAD

**Discussion by dm7500** [paperless-ngx/paperless-ngx#389] [Feature Requests]
Currently, Paperless-NG allows us to move emails with an attachment to a folder, and it will grab the attachment and save it. I'd really like to be able to do this for emails without an attachment as well, as they often contain useful information.

This has been brought up in the [-NG Github](https://github.com/jonaswinkler/paperless-ng/issues/319) as well.

**Comment by sense4t** [comment]
My workaround is printing emails to pdf and archiving these. In ios-mail this can be achieved easily with print-preview...

**Comment by KarstenDE** [comment]
I see this as useful. But how should paperless differentiate how to process mails? Especially if you forward the mails to a dedicated paperless-address?

One idea can be:
Configure multiple mail addresses in paperless and configure the processing-method.
Then you can say:
- paperless-mail@example.com >> just Archive the mail
- paperless-attachement@example.com >> just Archive the attachment(s)

What do you think?

**Reply by JanneW3** [reply]
On my phone I can use print as PDF and in my mail client, too. I think if NGX is able to link with an OneDrive folder, Google folder etc.  it's the best option.

**Comment by Sblop** [comment]
This ability is possible using container:

https://github.com/rob-luke/emails-html-to-pdf

I agree it would be nice if this was build in, however this works just fine.

**Reply by smseidl** [reply]
I am currently using this option for this issue.  You can add it into your same docker-compose.yml file and its likes its basically built into Paperless.

**Reply by mxmehl** [reply]
To confirm, that does not support saving the text/HTML part of an email *with* attachment, right?

**Comment by djusHa** [comment]
What about to archive the "email container", i mean eml or msg file, without parsing it. Meta / Index must be done manually. So if you download it, it can be opened with associated Application.

**Comment by mxmehl** [comment]
I wonder how this may relate to the missing feature of saving the email text alongside its attachments. Even if an email has an attachment, the body often contains important information that shall be saved alongside of it.

In other issues the idea of saving the email body text in an extra meta field has been brought up. I would also be fine with saving the email body text as a PDF and treating it as an attachment.

Anyway, I think it would make sense to tackle both issues – emails with attachment and without attachments – the same, and make the extraction and storage of the body text an optional function that can be enabled.

**Comment by stepanov1975** [comment]
Hi

I think it will be very useful to add both functionalities
1. To save email text together with the attachment
2. To save all email (as PDF or HTML)

**Comment by dmorawetz** [comment]
Another option considering #422 would be to save the email body and attachment(s) as separate documents and link them together in a group.

**Comment by p-h-a-i-l** [comment]
I'm currently working on archiving the mail as .eml alongside the extracted text in paperless. 
Curently I have a working prototype which can consume .eml files and extract them via imap.

**Reply by mxmehl** [reply]
Wow, that would be a great feature! How is your experience so far?

**Reply by p-h-a-i-l** [reply]
Still in Development, for productive use I need to finish some features first.

**Reply by p-h-a-i-l** [reply]
have a look at https://github.com/paperless-ngx/paperless-ngx/pull/848 to check the progress
